### DISCUSSION TITLE
Phone-Authentication android autoLogin

### DISCUSSION THREAD

**Discussion by alexookah** [capawesome-team/capacitor-firebase#251] [Q&A]
I recently switched to this awesome plugin. 
I would like to know if it's possible that android auto-logins with phone authentication without users needing to enter the received sms code. Does this work?

**Comment by robingenz** [comment]
Thank you!
There is Auto Fill if that's what you mean.
I have never tried this but maybe this [SO answer](https://stackoverflow.com/a/71600416/6731412) will help you.
Your message must be formatted correctly for this.

**Reply by alexookah** [reply]
I don't mean the autoFill. 
What I mean is when the sms is received in the android device it automatically completes the sign in flow, without users needing to read the sms code and enter it into the app to finish the login.

**Reply by alexookah** [reply]
In a [previous library I was using](https://github.com/baumblatt/capacitor-firebase-auth#notification-for-phone-verification-step) there was a callback that was called internally in the library where when the sms was received in the device it was logging in the user automatically (only on android).

**Reply by alexookah** [reply]
Maybe this is more related to this: https://stackoverflow.com/a/69302081/4334709

**Reply by alexookah** [reply]
I [investigated how the other library works](https://github.com/baumblatt/capacitor-firebase-auth/blob/fc1c7a458df718b67e64c19f0df9f7be175109a4/android/src/main/java/com/baumblatt/capacitor/firebase/auth/handlers/PhoneProviderHandler.java#L37) and it seems that the `PhoneAuthCredential has getSmsCode()`.

Firebase documentation: 
https://firebase.google.com/docs/reference/kotlin/com/google/firebase/auth/PhoneAuthCredential

Could we also use something like this?

**Reply by robingenz** [reply]
Thanks for investigation. This should be possible. I will have a look later.

**Reply by alexookah** [reply]
I will be happy to test it once implemented.

**Reply by robingenz** [reply]
I took a closer look and I think there is actually a bug in the current implementation.
However, I don't think this bug can be fixed without a breaking change.
I'll take a closer look next week and let you know when i found a fix (including support for `smsCode`) that can be tested.

**Reply by robingenz** [reply]
I will track any further progress on this discussion here: https://github.com/capawesome-team/capacitor-firebase/issues/252

**Reply by alexookah** [reply]
Tthank you very much for taking a look on this. I think it will give much value to users who login with phone and have android.
iOS does not have any similar issue to the bug that you found?

**Reply by robingenz** [reply]
No, because iOS solves this via apns notifications.

**Reply by alexookah** [reply]
Could you have a look on this android auto-login? 
It's the only feature I am missing to completely switch to this library.

**Reply by robingenz** [reply]
Yes, I will have a look at it this or next week. I am currently extremely busy.

**Reply by alexookah** [reply]
Did you have a chance to check it? Maybe you could describe the problem and I could make a PR if you like?

**Reply by robingenz** [reply]
Unfortunately not yet. I have a little more time next week. I am still not sure how to solve the problem and wanted to try a few things. It is important that no breaking change occurs. I'll have a look at it next week at the latest. v1.3.0 will be released this year and this issue is part of the version.

**Reply by iamrsojitra** [reply]
I am facing the same issue. It would be great if it's fixed.

**Reply by robingenz** [reply]
I have created a PR but haven't had time to test it yet: https://github.com/capawesome-team/capacitor-firebase/pull/272

If you want, you can help me with testing. You can find the latest dev build in the PR as a comment.

**Reply by iamrsojitra** [reply]
Ok let me test this

**Reply by iamrsojitra** [reply]
This is working fine for me.
I have tested this build 1.2.0-dev.4c00b61.1670180888

Thanks @robingenz

**Reply by alexookah** [reply]
@vc-rakesh I tried it but doesn't work. I even updated my production app with the dev build to see if it works only from google play install.
Could you maybe provide your code as an example?

**Reply by iamrsojitra** [reply]
Hey @alexookah 
I have just installed this build 1.2.0-dev.4c00b61.1670180888 and added one more event listener `phoneVerificationCompleted`

```
FirebaseAuthentication.addListener('phoneVerificationCompleted', (e) => {
        // The user is authenticated and you can handle login status via may be observable
});
```

**Reply by alexookah** [reply]
Did you check if the event.verificationCode is returned ?
I am using the verificationCode to complete the authentication.

**Reply by alexookah** [reply]
@robingenz @vc-rakesh  Do you know if the `skipNativeAuth` has to be `false` for the android auto login?

**Reply by iamrsojitra** [reply]
Hey @robingenz can you please deploy this build 1.2.0-dev.4c00b61.1670180888 to npm?

**Reply by robingenz** [reply]
@vc-rakesh One feature is still missing, then I release the new version. You can track it here: https://github.com/capawesome-team/capacitor-firebase/milestone/9

**Reply by alexookah** [reply]
Hey @robingenz . Sorry to bring this old discussion back. 
But do you know if android auto login works with skipNativeAuth: true? 
If not maybe it should be clarified somewhere in the docs.

**Reply by robingenz** [reply]
>  But do you know if android auto login works with skipNativeAuth: true?

@alexookah I have not tested it but it should work.

**Comment by iamrsojitra** [comment]


**Reply by robingenz** [reply]
Should work in both cases.

**Comment by gerardopar** [comment]
@robingenz hey hey, quick question. 

We currently have it so **SMS** logs in on both the native + web layer, which works great **_when a user enters the verification code manually_**, but we've run into an edge case where on **android** it could possibly get auto verified. is it possible to use the following event handler + login on the web layer. 

```javascript
FirebaseAuthentication.addListener('phoneVerificationCompleted', (e) => {
       // login on the web layer here
});
```

**Reply by robingenz** [reply]
Should work, but I can't say for sure right now as I don't have much time to look.
